src/vs/workbench/services/search/common/textSearchManager.ts

```diff
@@ -29,60 +29,73 @@ export class TextSearchManager {
 	constructor(private query: ITextQuery, private provider: TextSearchProvider, private fileUtils: IFileUtils, private processType: ITextSearchStats['type']) { }
 
 	search(onProgress: (matches: IFileMatch[]) => void, token: CancellationToken): Promise<ISearchCompleteStats> {
-		const folderQueries = this.query.folderQueries || [];
 		const tokenSource = new CancellationTokenSource(token);
 
 		return new Promise<ISearchCompleteStats>((resolve, reject) => {
 			this.collector = new TextSearchResultsCollector(onProgress);
 
 			let isCanceled = false;
-			const onResult = (result: TextSearchResult, folderIdx: number) => {
+			const onResult = (result: TextSearchResult) => {
 				if (isCanceled) {
 					return;
 				}
 
 				if (!this.isLimitHit) {
 					const resultSize = this.resultSize(result);
-					if (extensionResultIsMatch(result) && typeof this.query.maxResults === 'number' && this.resultCount + resultSize > this.query.maxResults) {
+					if (
+						extensionResultIsMatch(result) &&
+						typeof this.query.maxResults === 'number' &&
+						this.resultCount + resultSize > this.query.maxResults
+					) {
 						this.isLimitHit = true;
 						isCanceled = true;
 						tokenSource.cancel();
 
-						result = this.trimResultToSize(result, this.query.maxResults - this.resultCount);
+						result = this.trimResultToSize(
+							result,
+							this.query.maxResults - this.resultCount,
+						);
 					}
 
 					const newResultSize = this.resultSize(result);
 					this.resultCount += newResultSize;
 					if (newResultSize > 0 || !extensionResultIsMatch(result)) {
-						this.collector!.add(result, folderIdx);
+						this.collector!.add(result, 0); // Using 0 as a default folder index
 					}
 				}
 			};
 
-			// For each root folder
-			Promise.all(folderQueries.map((fq, i) => {
-				return this.searchInFolder(fq, r => onResult(r, i), tokenSource.token);
-			})).then(results => {
-				tokenSource.dispose();
-				this.collector!.flush();
-
-				const someFolderHitLImit = results.some(result => !!result && !!result.limitHit);
-				resolve({
-					limitHit: this.isLimitHit || someFolderHitLImit,
-					messages: flatten(results.map(result => {
-						if (!result?.message) { return []; }
-						if (Array.isArray(result.message)) { return result.message; }
-						else { return [result.message]; }
-					})),
-					stats: {
-						type: this.processType
-					}
-				});
-			}, (err: Error) => {
-				tokenSource.dispose();
-				const errMsg = toErrorMessage(err);
-				reject(new Error(errMsg));
-			});
+			// Single search call
+			const folderQuery = this.query.folderQueries?.[0] || {
+				folder: URI.file('/'),
+			};
```

Comment:
I'm concerned that we could be breaking some other functionality by ignoring the rest of the queries. Is there any way we can only do this when we're sure vscode is trying to search in _all_ directories?

It would be good to investigate if vscode uses this `search` functions for features other than `cmd+shift+f`.